### PR TITLE
Added `track_in_history`

### PR DESCRIPTION
**Pull request by biophyser**
Added in `track_in_history` functionality using json format instead of dataframes used in `v0.4.1`. Put the function in a separate module because I think it is a cool thing in its own right and keeps the `project.py` code cleaner.

### CODE REVIEW

**Comment by Zsailer** [comment]
This looks great! 

I need to test it still, but it looks good overall.

File: phylogenetics/history.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        time =strftime("%Y-%m-%d %H:%M:%S", localtime())
+
+        # Create a history item
+        history = {'time':time, 'method':str(method), 'args':str(args), 'kwargs':kwargs}
+
+        # Append to main history list
+        self.history.append(history)
+
+        # Write history to a json file.
+        history_file = os.path.join(self.project_dir, 'history.json')
```
Comment (Zsailer) [inline phylogenetics/history.py]: Lately, I've been on a [`pathlib`](https://docs.python.org/3/library/pathlib.html) kick. We may want to move away from `os.path` and just use `pathlib.Path` objects. 

It simplifies some of the logic and makes path handling more smooth.

This looks something like:
```python
import pathlib

# ...
# History as a string
history_json = json.dumps(self.history)

# Write history to file
history_file = pathlib.Path(self.project_dir).joinpath('history.json')
history_file.write_text(history_json)

**Comment by Zsailer** [comment]
We should probably add the `load` function here too. It needs to be away of the history file.

**Comment by biophyser** [comment]
Could you clarify your last comment? I get adding the `load` function but do you mean add it as a class function in `project.py`?

File: phylogenetics/history.py (hunk 1 above)
Comment (biophyser) [reply phylogenetics/history.py]: By simplify the logic do you mean makes it look more like `sklearn`? :stuck_out_tongue_winking_eye:

Comment (Zsailer) [reply phylogenetics/history.py]: Haha. I think the "write to file" logic is cleaner using `pathlib` than the typical syntax in Python:

```python
with open(history_file, 'w') as f:
    f.write(history_json)
```

**Comment by Zsailer** [comment]
Oops... there was a typo in my comment above -- the load function needs to be *aware* of the history file. What I mean is that the `load` function should read the project *and* history data (which live in two different files). 

As I'm thinking about this, I'm realizing the `load` function may need some extra discussion. Therefore, I propose we separate `load` into a separate pull request. I'm ready to merge if you're happy with it.

**Comment by biophyser** [comment]
That all makes sense. I realized in issue #15 that `load` depended on `track_in_history` decided to take that piecemeal. I think another PR for `load` is the way to go.

**Comment by Zsailer** [comment]
Merging

**Pull request merged**